src/main-process/win-shell.coffee

```diff
@@ -0,0 +1,53 @@
+Registry = require 'winreg'
+Path = require 'path'
+
+exeName = Path.basename(process.execPath)
+appPath = "\"#{process.execPath}\""
+appName = exeName.replace('atom', 'Atom').replace('beta', 'Beta').replace('.exe', '')
+
+class ShellOption
+  constructor: (key, parts) ->
+    @key = key
+    @parts = parts
+  isRegistered: (callback) =>
```

Comment:
The fat arrows are required so that the @ operator points to this object not to the window when the callback functions are executed.

I can add spacing.